packages/dev/serializers/src/glTF/2.0/Extensions/KHR_lights_punctual.ts

```diff
@@ -59,132 +69,113 @@ export class KHR_lights_punctual implements IGLTFExporterExtensionV2 {
+                      ? KHRLightsPunctual_LightType.DIRECTIONAL
+                      : babylonNode.getTypeID() == Light.LIGHTTYPEID_SPOTLIGHT
+                        ? KHRLightsPunctual_LightType.SPOT
+                        : null;
+            if (!lightType) {
+                Logger.Warn(`${context}: Light ${babylonNode.name} is not supported in ${NAME}`);
+                resolve(node);
+                return;
+            }
+
+            if (babylonNode.falloffType !== Light.FALLOFF_GLTF) {
+                Logger.Warn(`${context}: Light falloff for ${babylonNode.name} does not match the ${NAME} specification!`);
+            }
+
+            // Set the node's translation and rotation here, since lights are not handled in exportNodeAsync
+            if (!babylonNode.position.equalsToFloats(0, 0, 0)) {
+                const translation = TmpVectors.Vector3[0].copyFrom(babylonNode.position);
+                if (convertToRightHanded) {
+                    convertToRightHandedPosition(translation);
+                }
+                node.translation = translation.asArray();
+            }
+
+            // Babylon lights have "constant" rotation and variable direction, while
+            // glTF lights have variable rotation and constant direction. Therefore,
+            // compute a quaternion that aligns the Babylon light's direction with glTF's constant one.
+            if (lightType !== KHRLightsPunctual_LightType.POINT) {
+                const direction = babylonNode.direction.normalize();
+                if (convertToRightHanded) {
+                    convertToRightHandedPosition(direction);
+                }
+                const angle = Math.acos(Vector3.Dot(LIGHTDIRECTION, direction));
+                const axis = Vector3.Cross(LIGHTDIRECTION, direction);
+                const lightRotationQuaternion = Quaternion.RotationAxis(axis, angle);
+                if (!Quaternion.IsIdentity(lightRotationQuaternion)) {
+                    node.rotation = lightRotationQuaternion.normalize().asArray();
+                }
+            }
+
+            let light: IKHRLightsPunctual_Light = {
+                type: lightType,
+                name: babylonNode.name,
+                color: babylonNode.diffuse.asArray(),
+                intensity: babylonNode.intensity,
+                range: babylonNode.range,
+            };
+            light = omitDefaultValues(light, DEFAULTS);
+
+            // Separately handle the required 'spot' field for spot lights
+            if (lightType === KHRLightsPunctual_LightType.SPOT) {
+                const babylonSpotLight = babylonNode as SpotLight;
+                light.spot = {
+                    innerConeAngle: babylonSpotLight.innerAngle / 2.0,
+                    outerConeAngle: babylonSpotLight.angle / 2.0,
+                };
+                light.spot = omitDefaultValues(light.spot, SPOTDEFAULTS!);

```

Comment:
Should be able to remove `!` with the changes to the defaults.

```suggestion
                light.spot = omitDefaultValues(light.spot, SPOTDEFAULTS);
```

Reply:
Done